### ISSUE TITLE
Unset route query on default value is not working anymore

### ISSUE THREAD

**Issue by greenhalos** [vueuse/vueuse#3290]
## Setup

Package: @vueuse/router

```js
<script setup>
import { useRouteQuery } from "@vueuse/router";
const search = useRouteQuery("search", "defaultValue");
</script>
```

## Expected Beahviour and Behaviour of Version v10.2.1

When I change the value of `search` to the default value (in this case `"defaultValue"`), the query parameter is removed from the URL:
```js
search.value = "defaultValue";
```

## Behaviour in Version  v10.3.0

When changing the value of `search` to the default value (in this case `"defaultValue"`), the parameter remains in the URL.

## Further Notes
In my production use case the default value is `""` which leads to to a paramter `?search=`

## Further observations

* It was intended behaviour in the past, as far as I can tell: https://github.com/vueuse/vueuse/pull/1490
* It was remove in https://github.com/vueuse/vueuse/commit/0d02f6cc6a0a78374ab81e13f594f16eb4695b6a as part of https://github.com/vueuse/vueuse/pull/3184
* There is one Discussion about the same method, but is more about a changed behaviour https://github.com/vueuse/vueuse/discussions/2914

### Reproduction

https://github.com/greenhalos/useRouteQuery-bug-default

### System Info

```Shell
Binaries:
    Node: 18.15.0 - ~/.nvm/versions/node/v18.15.0/bin/node
    npm: 9.5.0 - ~/.nvm/versions/node/v18.15.0/bin/npm
    pnpm: 8.6.11 - ~/.local/share/pnpm/pnpm
  npmPackages:
    @vueuse/core: 10.3.0 => 10.3.0 
    @vueuse/router: 10.3.0 => 10.3.0 
    vue: ^3.2.25 => 3.3.4
```

### Used Package Manager

npm

### Validations

- [X] Follow our [Code of Conduct](https://github.com/vueuse/vueuse/blob/main/CODE_OF_CONDUCT.md)
- [X] Read the [Contributing Guidelines](https://github.com/vueuse/vueuse/blob/main/CONTRIBUTING.md).
- [X] Read the [docs](https://vueuse.org/guide).
- [X] Check that there isn't [already an issue](https://github.com/vueuse/vueuse/issues) that reports the same bug to avoid creating a duplicate.
- [X] Make sure this is a VueUse issue and not a framework-specific issue. For example, if it's a Vue SFC related bug, it should likely be reported to https://github.com/vuejs/core instead.
- [X] Check that this is a concrete bug. For Q&A open a [GitHub Discussion](https://github.com/vueuse/vueuse/discussions).
- [X] The provided reproduction is a [minimal reproducible example](https://stackoverflow.com/help/minimal-reproducible-example) of the bug.

**Comment by vuthanhbayit** [comment]
I just want to ask, when I use:

const state = reactive({
  name: useRouteQuery('name', ''),
})

watch(state, (newValue, oldValue) => {
  console.log(newValue === oldValue)
  console.log(newValue.name === oldValue.name)
})

when changing the value of state.name, console.log in watch returns true. 
How does it work? Tks for watching

**Comment by greenhalos** [comment]
It is fixed via #3583
